### DISCUSSION TITLE
Create sfdx file namespace commands

### DISCUSSION THREAD

**Discussion by AllanOricil** [forcedotcom/cli#2344] [Ideas]
**Is your feature request related to a problem? Please describe.**

Nope. I just thought that since we have `sfdx data` we could also have `sfdx file` to work with all types of file storage abstractions Salesforce has, such as contentDocuments, attachments and libraries.

**What are you trying to do**
Upload, download, delete files to/from Salesforce

**Describe the solution you'd like**

There could exist a new namespace command called file/doc for uploading/downloading/deleting all types of files Salesforce has, like contentDocument and attachments

`sfdx file:upload|download|delete`

This command must work with streams if the API supports it... we don't want to load an entire file in memory just to start uploading it...
This command should work with multiple files.
Flags would control the type of file storage/api  that should be used.

There could also exist a command to display how much file storage was used in a particular salesforce  instance.

**Describe alternatives you've considered**
Use 3rd party solutions or create my own integration

**Additional context**
N/A

[a comment by an automated account is not shown]

**Comment by AllanOricil** [comment]
https://github.com/forcedotcom/cli/issues/1993

**Comment by iowillhoit** [comment]
Thanks @AllanOricil, it's in our backlog 👍

**Comment by jshackell-sfdc** [comment]
This issue is addressed in version 2.44.8 (Jun 5, 2024)
